pennylane/interfaces/jacobian_products.py

```diff
@@ -0,0 +1,173 @@
+
+        Returns:
+            ResultBatch, TensorLike
+
+        **Examples:**
+
+        >>> tape0 = qml.tape.QuantumScript([qml.RX(0.1, wires=0)], [qml.expval(qml.PauliZ(0))])
+        >>> tape1 = qml.tape.QuantumScript([qml.RY(0.2, wires=0)], [qml.expval(qml.PauliZ(0))])
+        >>> batch = (tape0, tape1)
+        >>> tangents0 = (1.5, )
+        >>> tangents1 = (2.0, )
+        >>> tangents = (tangents0, tangents1)
+        >>> results, jvps = jp_method.execute_and_compute_jvp(batch, tangents)
+        >>> results
+        (0.9950041652780258, 0.9800665778412417)
+        >>> np.cos(0.1), np.cos(0.2)
+        (0.9950041652780258, 0.9800665778412416)
+        >>> jvps
+        (array(-0.14975012), array(-0.39733866))
+        >>> 1.5 * -np.sin(0.1), 2.0 * -np.sin(0.2)
+        (-0.14975012497024223, -0.39733866159012243)
+
+        """
+
+    @abc.abstractmethod
+    def compute_vjp(self, tapes: Batch, dy: Tuple[Tuple[TensorLike]]) -> Tuple:
+        """Compute the vjp for a given batch of tapes.
+
+        This method is used by autograd, torch, and tensorflow.
+
+        Args:
+            tapes: the batch of tapes to the the derivatives of
+            dy: the derivatives of the results of an execution
+
+        Returns:
+            TensorLike
+
+        >>> tape0 = qml.tape.QuantumScript([qml.RX(0.1, wires=0)], [qml.expval(qml.PauliZ(0))])
+        >>> tape1 = qml.tape.QuantumScript([qml.RY(0.2, wires=0)], [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliX(0))])
+        >>> batch = (tape0, tape1)
+        >>> dy0 = (0.5, )
+        >>> dy1 = (2.0, 3.0)
+        >>> dys = (dy0, dy1)
+        >>> jp_method.compute_vjp(batch, dys)
+        (array([-0.04991671]), array([2.54286107]))
+        >>> 0.5 * -np.sin(0.1)
+        -0.04991670832341408
+        >>> 2.0 * -np.sin(0.2) + 3.0 * np.cos(0.2)
+        2.5428610719336024
+
+        """
+
+    @abc.abstractmethod
+    def compute_jacobian(self, tapes: Batch) -> Tuple:
+        """Compute the full jacobian for a batch of tapes.
+
+        This method is required for jax-jit.
+
+        Args:
+            tapes: the batch of tapes to take the jacobian of
+
+        >>> tape0 = qml.tape.QuantumScript([qml.RX(0.1, wires=0)], [qml.expval(qml.PauliZ(0))])
+        >>> tape1 = qml.tape.QuantumScript([qml.RY(0.2, wires=0)], [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliX(0))])
+        >>> batch = (tape0, tape1)
+        >>> jp_method.compute_jacobian(batch)
+        (array(-0.09983342), (array(-0.19866933), array(0.98006658)))
+
+        """
+
+
+class TransformDerivatives(JacobianProductCalculator):
+    """Compute vjp, jvps, and jacobians via a gradient transform.
+
+    Args:
+        inner_execute (Callable[[Tuple[QuantumTape]], ResultBatch]): a function that
+            turns the batch of circuits into results.
+        gradient_transform (qml.gradients.gradient_transform): the gradient transform to use.
```

Comment:
docs link to this type isn't rendering

Reply:
Would this work, maybe?
```suggestion
        gradient_transform (`~.gradient_transform`): the gradient transform to use.
```